### ISSUE TITLE
Generic ref T converter is not being applied

### ISSUE THREAD

**Issue by ghost** [nim-lang/Nim#17010]
If this worked it would allow for a less redundant approach to replacing methods with https://github.com/yglukhov/iface.
See https://forum.nim-lang.org/t/7491#47561

### Example
```nim
type
  Expression = ref object
  Literal = ref object
  PlusExpr = ref object

converter toExpression[T](v: ref T): Expression = Expression()
proc newLit(x: int): Literal = discard
proc newPlus(a: Expression; b: Expression): PlusExpr = discard

discard newPlus(newPlus(newLit(1), newLit(2)), newLit(4).toExpression) # Works
discard newPlus(newPlus(newLit(1), newLit(2)), newLit(4)) # Doesn't work
```

### Current Output
```
/home/dian/Projects/nim-snippets/asd.nim(11, 16) Error: type mismatch: got <PlusExpr, Literal>
but expected one of: 
proc newPlus(a: Expression; b: Expression): PlusExpr
  first type mismatch at position: 2
  required type for b: Expression
  but expression 'newLit(4)' is of type: Literal

expression: newPlus(newPlus(newLit(1), newLit(2)), newLit(4))
```

### Expected Output
Successful compilation

```
$ nim -v
Nim Compiler Version 1.5.1 [Linux: amd64]
Compiled at 2021-02-10
Copyright (c) 2006-2021 by Andreas Rumpf

git hash: 216be4060a853b3425501318537d598c4842eefc
active boot switches: -d:release
```

**Comment by cooldome** [comment]
Known limitation. One generic converter can be instantiated only once per call. It was previously reported.

**Comment by cooldome** [comment]
Duplicate of #13843
